plugins/inputs/docker/docker.go

```diff
@@ -346,6 +346,39 @@ func (d *Docker) gatherInfo(acc telegraf.Accumulator) error {
 	return nil
 }
 
+func parseImage(image string) (string, string) {
+	// Adapts some of the logic from the actual Docker library's image parsing
+	// routines:
+	// https://github.com/docker/distribution/blob/release/2.7/reference/normalize.go
+	domain := ""
+	remainder := ""
+
+	i := strings.IndexRune(image, '/')
+
+	if i == -1 || (!strings.ContainsAny(image[:i], ".:") && image[:i] != "localhost") {
+		remainder = image
+	} else {
+		domain, remainder = image[:i], image[i+1:]
```

Comment:
Sorry, I'm not sure I understand. Image names like ``influxdata/`` or ``telegraf:`` aren't valid. If you ``docker tag <someimage> telegraf:`` you get a parsing error:

```
Error parsing reference: "telegraf:" is not a valid repository/tag: invalid reference format
```

Reply:
[This](https://github.com/docker/distribution/blob/release/2.7/reference/normalize.go#L64) is where I got this conditional from. This function is how Docker itself parses the domain component of an image name.

Reply:
Just trying to come up with some edge cases, even with unexpected data, since the input could be anything an we would never want to panic.  Looking closer it looks okay since you can always create a slice starting any any valid index+1.  I could imagine a case for an error for bad input, for example the `telegraf:` test gives an empty string version instead of unknown, but its not required.